pkg/devfile/parser/parse.go

```diff
@@ -392,10 +397,38 @@ func parseFromURI(importReference v1.ImportReference, curDevfileCtx devfileCtx.D
 	if !absoluteURL && curDevfileCtx.GetAbsPath() != "" {
 		newUri = path.Join(path.Dir(curDevfileCtx.GetAbsPath()), uri)
 		d.Ctx = devfileCtx.NewDevfileCtx(newUri)
+		if !strings.HasSuffix(newUri, ".yaml") {
+			return DevfileObj{}, fmt.Errorf("the provided path is not a valid yaml filepath: %s", newUri)
+		}
+		if util.ValidateFile(newUri) != nil {
+			return DevfileObj{}, fmt.Errorf("failed to read devfile from path %s", newUri)
+		}
+		srcDir := path.Dir(newUri)
+		destDir := path.Dir(curDevfileCtx.GetAbsPath())
+		if srcDir != destDir {
+			err := util.CopyAllDirFiles(srcDir, destDir)
+			if err != nil {
+				return DevfileObj{}, err
+			}
+		}
 	} else if absoluteURL {
 		// absolute URL address
 		newUri = uri
 		d.Ctx = devfileCtx.NewURLDevfileCtx(newUri)
+		if strings.Contains(newUri, "raw.githubusercontent.com") {
+			urlComponents, err := util.GetGitUrlComponentsFromRaw(newUri)
+			if err != nil {
+				return DevfileObj{}, err
+			}
+			if !strings.Contains(urlComponents["file"], "devfile.yaml") {
```

Comment:
you don't need to check the devfile name here. can be any name, as long as it contains a valid devfile content. and the parser will error out later if it is not a valid devfile

Reply:
removed